cl/search/management/commands/cl_index_parent_and_child_docs.py

```diff
@@ -1,14 +1,67 @@
-from typing import Iterable
+from datetime import datetime
+from typing import Iterable, Mapping
 
 from django.conf import settings
 
 from cl.lib.celery_utils import CeleryThrottle
 from cl.lib.command_utils import VerboseCommand
+from cl.lib.redis_utils import make_redis_interface
 from cl.people_db.models import Person
 from cl.search.models import SEARCH_TYPES, Docket
 from cl.search.tasks import index_parent_and_child_docs
 
 
+def compose_redis_key(search_type: str) -> str:
+    """Compose a Redis key based on the search type for indexing log.
+
+    :param search_type: The type of search.
+    :return: A Redis key as a string.
+    """
+    return f"es-{search_type}_indexing:log"
+
+
+def log_last_parent_document_processed(
+    search_type: str, document_pk: int
+) -> Mapping[str | bytes, int | str]:
+    """Log the last document_id indexed.
+
+    :param search_type: The search type key to log.
+    :param document_pk: The last document_id processed.
+    :return: The data logged to redis.
+    """
+
+    r = make_redis_interface("STATS")
+    pipe = r.pipeline()
+    log_key = compose_redis_key(search_type)
+    pipe.hgetall(log_key)
+    log_info: Mapping[str | bytes, int | str] = {
+        "last_document_id": document_pk,
+        "date_time": datetime.now().isoformat(),
+    }
+    pipe.hset(log_key, mapping=log_info)
+    pipe.expire(log_key, 60 * 60 * 24 * 28)  # 4 weeks
+    pipe.execute()
+
+    return log_info
+
+
+def get_last_parent_document_id_processed(search_type: str) -> int:
+    """Get the last document ID indexed.
+
+    :param search_type: The search type key to get the last document ID.
+    :return: The last document ID indexed.
+    """
+
+    r = make_redis_interface("STATS")
```

Comment:
I'd use the `CACHE` DB instead, since this is just a temporary thing.

Reply:
Sure, I've changed it to `CACHE`